runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillStateInternals.java

```diff
@@ -474,9 +474,12 @@ private WindmillValue(
 
     @Override
     public void clear() {
-      modified = true;
+      // if the value was already null (because it was already cleared) then there's no need to
+      // mark this as modified again.  This will save having to persist a clear that does nothing.
```

Comment:
this change breaks a couple unit tests that were asserting the commit payload looked a certain way, but I'm don't think the change is actually wrong.  The state being cleared was never persisted, so clearing it again is a noop.  I'll update the unit tests.